pkg/tests/end_to_end_tests/telemetry_test.go

```diff
@@ -116,12 +116,92 @@ func TestOnlyOneHousekeeper(t *testing.T) {
+
+		var val string
+		err = conn.QueryRow(context.Background(), "select value from _timescaledb_catalog.metadata where key = 'promscale_ingested_samples_total'").Scan(&val)
 		require.Error(t, err)
-		require.True(t, strings.Contains(err.Error(), `violates unique constraint "promscale_instance_information_pkey"`))
+		require.Equal(t, "no rows in result set", err.Error())
+
+		mockMetric.Add(100)
+
+		require.NoError(t, engine.Sync())
+
+		err = conn.QueryRow(context.Background(), "select value from _timescaledb_catalog.metadata where key = 'promscale_ingested_samples_total'").Scan(&val)
+		require.NoError(t, err)
+		require.Equal(t, "100", val)
+	})
+}
+
+func TestCleanStalePromscales(t *testing.T) {
+	withDB(t, *testDatabase, func(dbOwner *pgxpool.Pool, t testing.TB) {
+		db := testhelpers.PgxPoolWithRole(t, *testDatabase, "prom_writer")
+		defer db.Close()
+
+		conn := pgxconn.NewPgxConn(db)
+
+		engine, err := telemetry.NewEngine(conn, generateUUID())
+		require.NoError(t, err)
+
+		engine.Start()
+		defer engine.Stop()
+
+		var cnt int64
+		err = conn.QueryRow(context.Background(), "SELECT count(*) FROM _ps_catalog.promscale_instance_information").Scan(&cnt)
+		require.NoError(t, err)
+		require.Equal(t, 1, int(cnt)) // The counter reset row.
+
+		// Insert a stale Promscale instance row.
+		_, err = conn.Exec(context.Background(), `INSERT INTO _ps_catalog.promscale_instance_information (uuid, last_updated, promscale_ingested_samples_total, is_counter_reset_row) VALUES ('10000000-0000-0000-0000-000000000000', current_timestamp - INTERVAL '1 DAY', 100, FALSE)`)
+		require.NoError(t, err)
+
+		err = conn.QueryRow(context.Background(), "SELECT count(*) FROM _ps_catalog.promscale_instance_information").Scan(&cnt)
+		require.NoError(t, err)
+		require.Equal(t, 2, int(cnt)) // The counter reset row + promscale instance row added above.
+
+		err = conn.QueryRow(context.Background(), "SELECT promscale_ingested_samples_total FROM _ps_catalog.promscale_instance_information WHERE is_counter_reset_row = TRUE").Scan(&cnt)
+		require.NoError(t, err)
+		require.Equal(t, 0, int(cnt))
+
+		exists := false
+		err = conn.QueryRow(context.Background(), "SELECT count(*) > 0 FROM _ps_catalog.promscale_instance_information WHERE uuid = '10000000-0000-0000-0000-000000000000'").Scan(&exists)
+		require.NoError(t, err)
+		require.True(t, exists)
+
+		// Clean up stale promscale rows.
+		require.NoError(t, engine.Sync())
+
+		err = conn.QueryRow(context.Background(), "SELECT count(*) > 0 FROM _ps_catalog.promscale_instance_information WHERE uuid = '10000000-0000-0000-0000-000000000000'").Scan(&exists)
+		require.NoError(t, err)
+		require.False(t, exists)
+
+		// Check counter reset row's value.
+		err = conn.QueryRow(context.Background(), "SELECT promscale_ingested_samples_total FROM _ps_catalog.promscale_instance_information WHERE is_counter_reset_row = TRUE").Scan(&cnt)
+		require.NoError(t, err)
+		require.Equal(t, 100, int(cnt))
```

Comment:
let's run `require.NoError(t, engine.Sync())` here again to make sure the logic is safe when the timeout hasn't. been reached yet.